### PR TITLE
Platform independent test script, #195

### PR DESCRIPTION
**Pull request by erikbarke**

### CODE REVIEW

File: ci/run.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+var spawn = require("cross-spawn");
+var os = require("os");
+
+var browser = os.platform() === "win32" ? "" : ":phantomjs";
+
+// The pyramid of doom :( If you have a better idea, (which doesn't
```
Comment (mitchhentges) [inline ci/run.js]: Pyramid of doom haha!
Theoretically, we can even run these in parallel, so when a solution is created to clean up the pyramid, it also has the opportunity of super-improving the test speed

File: ci/run.bat [hunk 2]
```diff
@@ -1,27 +0,0 @@
-cd examples\angular2
```
Comment (mitchhentges) [inline ci/run.bat]: YASS

File: appveyor.yml [hunk 3]
```diff
@@ -12,6 +12,6 @@ test_script:
   - npm run lint
   - npm run build
   - npm run test:unit
-  - ci\run.bat
+  - npm test
```
Comment (mitchhentges) [inline appveyor.yml]: 👌 🥇

File: ci/run.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+// involve wait.for / node-gyp / fibers) please create a pull request,
+// but bear in mind that this has to run on windows machines as well...
+execute("examples/angular2", "npm run dev:ci" + browser, function() {
```
Comment (mitchhentges) [inline ci/run.js]: The `npm run dev:ci` string is copied, so you could pull that into a separate constant variable I suppose, maybe

File: ci/run.js (hunk 1 above)
Comment (erikbarke) [reply ci/run.js]: Yes, they run quite smoothly in parallel, but the logging gets all screwed up, it gets hard to tell which logging statement belongs to which project...

File: ci/run.js (hunk 4 above)
Comment (erikbarke) [reply ci/run.js]: Needs a bit of cleanup :)

File: ci/run.js (hunk 1 above)
Comment (mitchhentges) [reply ci/run.js]: True, though we could do some sort of buffering - [lerna](https://github.com/lerna/lerna) can run commands concurrently but somehow produces useful logs. Either way, not a priority, but a potential improvement :)

Comment (erikbarke) [reply ci/run.js]: The pyramid looks a bit like ascii art, doesn't it 🤓

**Pull request merged**